registries/adapters/adapter.go

```diff
@@ -57,21 +58,32 @@ type Configuration struct {
 	Tag        string
 }
 
-// Retrieve the spec from a registry manifest request
-func imageToSpec(req *http.Request, image string) (*bundle.Spec, error) {
-	log.Debug("Registry::imageToSpec")
-	spec := &bundle.Spec{}
-	req.Header.Add("Accept", "application/json")
-
-	resp, err := http.DefaultClient.Do(req)
+func commonResponseHandler(resp *http.Response) ([]byte, error) {
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusUnauthorized {
+		log.Errorf("Unable to authenticate to the registry, registry credentials could be invalid.")
+		return nil, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		log.Errorf("unexpected response code %v body %v", resp.StatusCode, string(body))
+		return nil, nil
+	}
```

Comment:
So `!StatusOK` and `StatusUnauthorized` are *NOT* error conditions? seems like these should return some sort of error. i.e. `return nil, fmt.Error("Unauthorized")` vs nil.

Reply:
Had the same question independently. AFAICT these were refactored from previous code. I'm wondering if maybe we should return errors here, and the calling code can choose to care about it.

Reply:
+1 I would prefer to see an error here especially on an Unauthorized. This way as we begin to refactor to the adapters we can have some sanity checks using the `commonResponseHandler` with grabbing auth tokens.

Reply:
As @eriknelson mentioned, this code was just moved it functions as it did before. I am happy to update to the suggested approach though